app/controllers/calculators/take_whole_pot_controller.rb

```diff
@@ -5,6 +5,10 @@ def show
       @income = params[:income]
 
       @calculator = TakeWholePotCalculatorForm.new(pot: @pot, income: @income)
+
+      render partial: 'calculators/take_whole_pot/calculator',
+             locals: { calculator: @calculator, pot: @pot, income: @income },
+             status: (@calculator.invalid? ? 400 : 200) if request.xhr?
```

Comment:
[Rails understands both numeric status codes and corresponding symbols](http://guides.rubyonrails.org/layouts_and_rendering.html).

I think using symbols improves the readability if you're not hot on your HTTP status codes.

```rb
status: (@calculator.valid? ? :ok : :bad_request)
```